Improved the devtools pages (#162)
rsksmart/devportal: docs/05-dev-tools/node-rpc/getBlock.md

---
sidebar_position: 3
title: GetBlock

> sidebar_label: getBlock

Reviewer: ```suggestion
sidebar_label: GetBlock
```

## Why Choose GetBlock?  
- **Broad Blockchain Support:** Access Ethereum, Bitcoin, Binance Smart Chain, Solana, and more.  
- **High Availability:** Enterprise-grade reliability with 99.95% uptime.  
- **Scalable Solutions:** Pay-as-you-go plans tailored to your project size.  

## Supported Protocols  
GetBlock supports these protocols for interacting with blockchain nodes:  
- **JSON-RPC**  
- **REST**  
- **WebSocket**  

## Key Blockchain Networks  
- **Ethereum** (`eth-mainnet`, `eth-goerli`)  
- **Bitcoin** (`btc-mainnet`, `btc-testnet`)  
- **Binance Smart Chain** (`bsc-mainnet`, `bsc-testnet`)  
- **Polygon** (`matic-mainnet`, `matic-testnet`)  

## **Getting Started with GetBlock API**  

### 1. Sign Up and Get an access Token  
- Create an account at [GetBlock.io](https://getblock.io).  
- Generate an [access token](https://getblock.io/docs/get-started/auth-with-access-token/) from your dashboard.  

### 2. Configure Your Endpoint  
- Base URL:  
  ```
  https://{blockchain}.getblock.io/mainnet/YOUR_API_KEY/
  ```  

>   Replace `{blockchain}` with the desired blockchain name (e.g., `eth`).

Reviewer: ```suggestion
  Replace `{blockchain}` with the desired blockchain name (e.g., `rsk`).
```

## **Make Your First Call**  

1. Use a tool like Postman.  

> 2. Use the Ethereum endpoint as an example:  

Reviewer: ```suggestion
2. Use the Rootstock RPC endpoint as an example:  
```

   ```bash
   https://go.getblock.io/<ACCESS_TOKEN>/
   ```  
3. Send the following JSON request to fetch the current block number:  
   ```json
   {
     "jsonrpc": "2.0",
     "method": "eth_blockNumber",
     "params": [],
     "id": 1
   }
   ```  
4. You’ll receive a response like:  
   ```json
   {
     "jsonrpc": "2.0",
     "method": "eth_blockNumber",
     "params": [],
     "id": 1
   }
   ```  
   > The `result` field contains the current block number in hexadecimal format.  

:::note[Info]

 To learn more about the getBlock connection with Rootstock, including code snippets, RPC endpoints,  click the button below:
 

> <Button href="https://getblock.io/nodes/rsk/" align="left">Rootstock getBlock Website</Button>

Reviewer: ```suggestion
<Button href="https://getblock.io/nodes/rsk/" align="left">Rootstock GetBlock Website</Button>
```
